<!-- Register a depth camera with an RGB camera -->
<!-- TODO: Should be able to remap namespaces in nodelet load with current roscpp -->
<launch>

  <arg name="manager" />

  <arg name="respawn" default="false" />
  <arg     if="$(arg respawn)" name="bond" value="" />
  <arg unless="$(arg respawn)" name="bond" value="--no-bond" />

  <arg name="sw_registered_processing" default="true" />
  <arg name="hw_registered_processing" default="true" />

  <!-- TODO: Instead "rgb", "depth" and "depth_registered" should be remapped as
       necessary by top-level launch file -->
  <arg name="rgb"              default="rgb" />
  <arg name="depth"            default="depth" />
  <arg name="depth_registered" default="depth_registered" />
  <!-- For distinguishing multiple register/XYZRGB nodelets. Default fails if rgb
       or depth contains a namespace. -->
  <arg name="suffix" default="$(arg depth)_$(arg rgb)" />
  <!-- For filtering depth images, set depth_registered_filtered to the
       filtered depth image topic -->
  <arg name="depth_registered_filtered" default="depth_registered" />


  <!-- Registration nodelet, projecting depth to RGB camera -->
  <group if="$(arg sw_registered_processing)">
    <!-- <node pkg="nodelet" type="nodelet" name="register_$(arg suffix)"
          args="load depth_image_proc/register $(arg manager) $(arg bond)"
          respawn="$(arg respawn)"> -->
      <!-- Explicit topic remappings, shouldn't need all of these -->
      <!-- <remap from="rgb/camera_info"             to="$(arg rgb)/camera_info" />
      <remap from="depth/camera_info"           to="$(arg depth)/camera_info" />
      <remap from="depth/image_rect"            to="$(arg depth)/image_rect_raw" />
      <remap from="depth_registered/image_rect" to="$(arg depth_registered)/sw_registered/image_rect_raw" />
    </node> -->

    <!-- Publish registered XYZRGB point cloud with software registered input -->
    <node pkg="nodelet" type="nodelet" name="points_xyzrgb_sw_registered"
          args="load depth_image_proc/point_cloud_xyzrgb $(arg manager) $(arg bond)"
          respawn="$(arg respawn)">
      <!-- Explicit topic remappings, shouldn't need all of these -->
      <!--Subscrive Topics-->
      <!-- <remap from="rgb/image_rect_color"        to="$(arg rgb)/image_rect_color" /> -->
      <!-- <remap from="rgb/camera_info"             to="$(arg rgb)/camera_info" /> -->
      <!-- <remap from="depth_registered/image_rect" to="$(arg depth_registered_filtered)/sw_registered/image_rect_raw" /> -->
      <remap from="rgb/image_rect_color"        to="/ext/camera/rgb/image_rect_color"/>
      <remap from="rgb/camera_info"             to="/hsrb/head_rgbd_sensor/rgb/camera_info"/>
      <remap from="depth_registered/image_rect" to="/ext/camera/depth_registered/image_raw"/>
      <!--Publish Topics-->
      <remap from="depth_registered/points"     to="/ext/camera/depth_registered/points"/>

      <param name="queue_size" value="10" />  <!--Custom-->
    </node>

    <!-- Rectified depth image -->
    <!-- <node pkg="nodelet" type="nodelet" name="$(arg depth_registered)_sw_metric_rect"
      args="load depth_image_proc/convert_metric $(arg manager) $(arg bond)"
      respawn="$(arg respawn)">
      <remap from="image_raw" to="$(arg depth_registered)/sw_registered/image_rect_raw" />
      <remap from="image"     to="$(arg depth_registered)/sw_registered/image_rect" />
    </node> -->
  </group>

</launch>
